dns-discovery/src/test/java/io/crate/discovery/SrvUnicastHostsProviderTest.java

```diff
@@ -22,206 +22,114 @@
-                int portStart = address.indexOf(':');
-                int port;
-                if (portStart >= 0) {
-                    port = Integer.parseInt(address.substring(portStart + 1));
-                } else {
-                    port = 44300;
-                }
-                int num = Integer.parseInt(address.substring("crate".length(), address.indexOf(".")));
-                byte[] dummyIp = new byte[] { 0, 0, 0, 0 };
-                try {
-                    return new TransportAddress[] {
-                        new TransportAddress(InetAddress.getByAddress("crate" + num + ".internal", dummyIp), port)
-                    };
-                } catch (UnknownHostException e) {
-                    throw new RuntimeException(e);
-                }
-            }
-        };
+        TransportService transportService = MockTransportService.createNewService(
+            Settings.EMPTY, Version.CURRENT, threadPool, null);
+        srvUnicastHostsProvider = new SrvUnicastHostsProvider(Settings.EMPTY, transportService);
     }
 
     @After
-    public void releaseThreadPool() throws Exception {
+    public void releaseResources() throws Exception {
+        srvUnicastHostsProvider.close();
+        srvUnicastHostsProvider.eventLoopGroup().shutdownGracefully();
+        srvUnicastHostsProvider.eventLoopGroup().awaitTermination(30, TimeUnit.SECONDS);
         threadPool.shutdown();
         threadPool.awaitTermination(30, TimeUnit.SECONDS);
     }
 
     @Test
-    public void testInvalidResolver() throws Exception {
-        Settings.Builder builder = Settings.builder()
-            .put(SrvUnicastHostsProvider.DISCOVERY_SRV_RESOLVER.getKey(), "foobar.txt")
-            .put(SrvUnicastHostsProvider.DISCOVERY_SRV_QUERY.getKey(), "_crate._srv.foo.txt");
-        SrvUnicastHostsProvider unicastHostsProvider = new SrvUnicastHostsProvider(builder.build(), transportService);
-        assertNull(unicastHostsProvider.resolver);
-        assertNull(unicastHostsProvider.lookupRecords());
-    }
-
-    @Test
-    public void testValidResolver() throws Exception {
-        Settings.Builder builder = Settings.builder()
-            .put(SrvUnicastHostsProvider.DISCOVERY_SRV_RESOLVER.getKey(), "8.8.4.4");
-        SrvUnicastHostsProvider unicastHostsProvider = new SrvUnicastHostsProvider(builder.build(), transportService);
-        assertEquals("/8.8.4.4:53", ((SimpleResolver) unicastHostsProvider.resolver).getAddress().toString());
-    }
-
-    @Test
-    public void testValidResolverWithPort() throws Exception {
-        Settings.Builder builder = Settings.builder()
-            .put(SrvUnicastHostsProvider.DISCOVERY_SRV_RESOLVER.getKey(), "127.0.0.1:5353");
-        SrvUnicastHostsProvider unicastHostsProvider = new SrvUnicastHostsProvider(builder.build(), transportService);
-        assertEquals("/127.0.0.1:5353", ((SimpleResolver) unicastHostsProvider.resolver).getAddress().toString());
+    public void testParseAddress() {
```

Comment:
I think it would be cool to have 3 tests instead of one that tests multiple scenarios.